### PR TITLE
Fix Symfony deprecations

### PR DESCRIPTION
**Pull request by blankse**
## Changes in this pull request  
Resolves
```
Since symfony/framework-bundle 6.1: Option "reset_on_message" at "framework.messenger" is deprecated. It does nothing and will be removed in version 7.0.
Since symfony/framework-bundle 6.1: Not setting the "framework.http_method_override" config option is deprecated. It will default to "false" in 7.0.
Since symfony/security-bundle 6.2: The "enable_authenticator_manager" option at "security" is deprecated.
User Deprecated: Since symfony/http-kernel 6.3: Parameter "container.dumper.inline_class_loader" is deprecated, use ".container.dumper.inline_class_loader" instead.
The "Pimcore\Serializer\Normalizer\ReferenceLoopNormalizer::supportsNormalization()" method will require a new "array $context" argument in the next major version of its interface "Symfony\Component\Serializer\Normalizer\NormalizerInterface", not defining it is deprecated.
```

### <samp>🤖 Generated by Copilot at 643af7f</samp>

This pull request improves security and performance of the pimcore application by disabling the HTTP method override feature, removing obsolete options from the configuration files, and refactoring the authentication system. It also updates the upgrade notes to document the breaking changes and the migration steps.

### <samp>🤖 Generated by Copilot at 643af7f</samp>

> _No more old authentication, no more override_
> _We purge the obsolete, we refactor and rewrite_
> _Breaking changes in the code, we document and advise_
> _We secure and optimize, we pimcore with pride_

### <samp>🤖 Generated by Copilot at 643af7f</samp>

*  Refactor and clean up the authentication system in pimcore ([link](https://github.com/pimcore/pimcore/pull/15796/files?diff=unified&w=0#diff-ee8ff688fded8943af83197fb3b0953bf27ea9ff986d6754f5c4d1a5b5767a3eL2-L3))
   - Update the upgrade notes in `README.md` to inform the users of the breaking change and the migration steps ([link](https://github.com/pimcore/pimcore/pull/15796/files?diff=unified&w=0#diff-375aded12dfb868b84d5b8c27d1aed2358680a27b683e8a985f1b85193dd79aaL85-R85))
* Disable the HTTP method override feature in Symfony for security and compatibility reasons ([link](https://github.com/pimcore/pimcore/pull/15796/files?diff=unified&w=0#diff-34d259c58a9001df91ca2ccc5f41ab960a307d1566cffceb88262d679cfc6c70R26))
* Remove the `reset_on_message` option from `default.yaml`, as it is not needed anymore ([link](https://github.com/pimcore/pimcore/pull/15796/files?diff=unified&w=0#diff-34d259c58a9001df91ca2ccc5f41ab960a307d1566cffceb88262d679cfc6c70L51))

### CODE REVIEW

File: lib/Controller/ArgumentValueResolver/WebsiteConfigValueResolver.php [hunk 1]
```diff
@@ -18,22 +18,24 @@
...
-use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
 use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
 
 /**
  * Adds support for type hinting controller actions and getting the current website config.
- *
  */
-final class WebsiteConfigValueResolver implements ArgumentValueResolverInterface
+final class WebsiteConfigValueResolver implements ValueResolverInterface
```
Comment (dvesh3) [inline lib/Controller/ArgumentValueResolver/WebsiteConfigValueResolver.php:29]: this class not marked as internal so removing a method is considered a BC break.

Comment (dvesh3) [reply lib/Controller/ArgumentValueResolver/WebsiteConfigValueResolver.php:29]: Ok. regression of https://github.com/pimcore/pimcore/pull/12911/files#diff-0a9c0cad482ae4e69e0fb9297c5d4e613892cd9cd09e93a68554a150c2c9abccL24. maybe we should mark it as internal again.

Comment (blankse) [reply lib/Controller/ArgumentValueResolver/WebsiteConfigValueResolver.php:29]: @dvesh3 Mark it again https://github.com/pimcore/pimcore/pull/15796/commits/f7cad40f0541168c961d74ce61c43a9b02cfe9be

**Comment by dvesh3** [comment]
@blankse thanks!

**Pull request merged**